test/bc_coeff_compositions.jl

```diff
@@ -433,25 +445,63 @@ end
     analytic_Qb = [zeros(N+1); sin(0.2+dx)]
     analytic_Ab = analytic_L*analytic_Qb
 
-    analytic_u = analytic_AL \ (u - analytic_Ab)
+
+    analytic_QM = zeros((length(x)+2)*3, length(x)*3)
+    interior = CartesianIndices(Tuple([2:length(x)+1, 1:3]))
+    I1 = CartesianIndex(1,0)
+    for I in interior
+        i = c2l(I, (length(x)+2, 3)) #helper function, see utils.jl
+        j = c2l(I-I1, (length(x), 3))
+        analytic_QM[i,j] = 1.0
+    end
+    analytic_Am = kron(Diagonal(ones(3)), analytic_L)*analytic_QM
+
+    analytic_u = analytic_AL \ Vector(u .- analytic_Ab)
     ghost_u = A \ u
 
-    # Check that A\u.(x) is consistent with analytic_AL \ u.(x)
-    @test analytic_u ≈ ghost_u
+    As_l, As_b = sparse(A, length(u))
 
-    # Check ldiv!
-    u_temp = zeros(N)
-    ldiv!(u_temp, A, u)
-    @test u_temp ≈ ghost_u ≈ analytic_u
+    @test As_l ≈ analytic_AL #This is true
+    @test As_b ≈  analytic_Ab #This is true
+    @test Vector(u .- analytic_Ab) ≈ Vector(u .- As_b) #this is true
+    u_translated = u .- analytic_Ab |> Vector
+    # TODO validate analytic_AL as having reasonable condition (what is reasonable condition?)
+    @test norm(analytic_AL \ u_translated - As_l \ u_translated) / norm(analytic_AL \ u_translated) < cond(analytic_AL)*eps()
+
+    sp_u = As_l\Vector(u .- As_b)
+    # Check that A\u.(x) is consistent with analytic_AL \ u.(x)
+    @test ghost_u ≈ sp_u
+    # TODO make cond function in ghost_derivative_operator?
+    # however, note: ArgumentError: 2-norm condition number is not implemented for sparse matrices, try cond(Array(A), 2) instead
+    # so the actual function would probably just propagate that error
+    _cond(A::GhostDerivativeOperator, u) = cond(sparse(A,length(u))[1] |> Array)
+    @test norm(analytic_u - ghost_u) / norm(ghost_u) < _cond(A,u) * eps() #
 
-    # Check \ for Matrix
     M2 = [u 2.0*u 10.0*u]
-    analytic_M = analytic_AL \ (M2 .- analytic_Ab)
-    ghost_M = A \ M2
-    @test analytic_M ≈ ghost_M
-
-    # Check ldiv! for Matrix
-    M_temp = zeros(N,3)
-    ldiv!(M_temp, A, M2)
-    @test M_temp ≈ ghost_M ≈ analytic_M
+    s = size(M2)
+    Qx = MultiDimBC{1}(Q, size(M2))
+    Am = L*Qx
+    #Somehow the operator is singular
+    @test_broken analytic_M = analytic_Am \ (reshape(M2, prod(s)) .- repeat(analytic_Ab, 3))
+    @test_broken ghost_M = Am \ M2
+    @test_broken reshape(analytic_M, s) ≈ ghost_M
+
+end
+
+@testset "Test Operator and BC combinations" begin
+    N = 40
+    x = range(-pi, stop = pi, length=N)
+    Δx = x[2]-x[1]
+    u₀=1.0
+    Γ=1.0
+    Dx=u₀*CenteredDifference{1}(1,4,Δx,N)
+    Dxx=Γ*CenteredDifference{1}(2,4,Δx,N)
+    Q=NeumannBC((2x[1]+2, 2x[end]+2), Δx, 6)
+
+    A = Dx*Q + Dxx*Q
+
+    y = A*(x.^2)
+    analytic_y = 2x.+2
+
+    @test_broken y ≈ analytic_y

```

Comment:
hmm, does it get close?

Reply:
```
julia> maximum(abs.(y .- analytic_y))
4.665789514226372
```

Reply:
oof okay that's an issue, but it can be skipped for now.